src/Orleans/AssemblyLoader/AssemblyCatalog.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Orleans.Runtime
+{
+    public class AssemblyCatalog : IAssemblyCatalog
```

Comment:
Sure. The way it is today may not be the 10th wonder now but it does what is suppose to do. So I think we can enhance it later. What matter is the interface and its usage. The catalog implementation itself can change anytime.